embedded/store/tx.go

```diff
@@ -67,37 +62,21 @@ func NewTxWithEntries(entries []*TxEntry) *Tx {
 	htree, _ := htree.New(len(entries))
 
 	return &Tx{
-		ID:       0,
-		entries:  entries,
-		nentries: len(entries),
-		htree:    htree,
+		header:  &TxHeader{NEntries: len(entries)},
+		entries: entries,
+		htree:   htree,
 	}
 }
 
 func (tx *Tx) Header() *TxHeader {
-	return &TxHeader{
-		ID:       tx.ID,
-		PrevAlh:  tx.PrevAlh,
-		Ts:       tx.Ts,
-		Metadata: tx.Metadata,
-		NEntries: tx.nentries,
-		Eh:       tx.Eh(),
-		BlTxID:   tx.BlTxID,
-		BlRoot:   tx.BlRoot,
-	}
+	return tx.header
 }
 
 func (hdr *TxHeader) Bytes() []byte {
-	// ID + PrevAlh + Ts + MDLen + MD + NEntries + Eh + BlTxID + BlRoot
-	var b [txIDSize + sha256.Size + tsSize + sszSize + maxTxMetadataLen + sszSize + sha256.Size + txIDSize + sha256.Size]byte
+	// ID + Ts + PrevAlh + Version + MDLen + MD + NEntries + Eh + BlTxID + BlRoot
+	var b [txIDSize + sha256.Size + tsSize + sszSize + (sszSize + maxTxMetadataLen) + sszSize + sha256.Size + txIDSize + sha256.Size]byte
```

Comment:
Low-prio (not a blocker): Isn't this code very simillar to ImmuStore.performCommit ?

Reply:
in some way yes, here the entries are not serialized but the root of the hash tree of the entries is included